### PR TITLE
Ports action figures

### PR DESCRIPTION
**Pull request by Niknakflak**
-Ports action figures from NT code
-Minor code edits from stuff that no longer worked
-Feel free to suggest other/better phrases for them to yell
-Did not include the action figures winnable in the arcades because I feel it would be spammy and everyone would win action figures instead of any of the other toys anymore, feel free to suggest somewhere else they can be used as well
-Action figures are good for admin spawned/made chess as we used to play on NT

Original PR: https://github.com/NTStation/NTstation13/pull/728

### CODE REVIEW

**Comment by Bawhoppen** [comment]
niknak you never play chess with me anymore

**Comment by Niknakflak** [comment]
These weren't in game so what else would we use, this makes it possible again

**Comment by Bawhoppen** [comment]
recolored mechs that take 10 minutes to set up?

**Comment by Niknakflak** [comment]
too much work, this makes it easier anyway

**Comment by Tokiko1** [comment]
Couldn't you just make an action figure item that generates a random action figure upon spawning and add that to the arcade prize list?

**Comment by Bawhoppen** [comment]
don't we still need to recolor them for each side? even if they are thematically correct for their roles now

**Comment by MrSnapwalk** [comment]
Oh hey, something I did back in the grand old days of the mighty NTstation.

Here's a picture of all of them (probably).

(also I beat you to it over on Bay, haha: Baystation12/Baystation12/pull/8158)

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1212,4 +1212,251 @@
...
+	name = "Non-Specific Action Figure action figure"
+	desc = "A \"Space Life\" brand... wait, what the hell is this thing?"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "nuketoy"
+	var/cooldown = 0
+	var/toysay = "What the fuck did you do?"
+	var/obj/item/toy/accessory/hastoy = null
+
+/obj/item/toy/figure/attack_self(mob/user as mob)
+	if(!cooldown)
```
Comment (Jordie0608) [inline code/game/objects/items/toys.dm]: Use world.time for cooldowns

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1212,4 +1212,251 @@
...
+
+
+
+/*
+ * Action Figures
+ */
+
+/obj/item/toy/figure
+	name = "Non-Specific Action Figure action figure"
+	desc = "A \"Space Life\" brand... wait, what the hell is this thing?"
```
Comment (Jordie0608) [inline code/game/objects/items/toys.dm]: Remove all the desc defined in the children and replace this with:
```
desc = "A \"Space Life\" brand [name]."
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1212,4 +1212,251 @@
...
+	var/obj/item/toy/accessory/hastoy = null
+
+/obj/item/toy/figure/attack_self(mob/user as mob)
+	if(!cooldown)
+		user << "<span class='notice'>The [src] says \"[toysay]\".</span>"
+		playsound(user, 'sound/machines/click.ogg', 20, 1)
+		cooldown = 1
+		spawn(30) cooldown = 0
+
+/*/obj/item/toy/figure/attack_hand(mob/user as mob)
```
Comment (Jordie0608) [inline code/game/objects/items/toys.dm]: Don't add in commented out code

**Comment by Bawhoppen** [comment]
I don't think niknak coded it. ALSO MAGNIFYING GLASSES WHEN AND MELTING ACTION FIGURES AND SIPPING THE WAX WHEN

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1212,4 +1212,251 @@
...
+ */
+
+/obj/item/toy/figure
+	name = "Non-Specific Action Figure action figure"
+	desc = "A \"Space Life\" brand... wait, what the hell is this thing?"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "nuketoy"
+	var/cooldown = 0
+	var/toysay = "What the fuck did you do?"
+	var/obj/item/toy/accessory/hastoy = null
```
Comment (Jordie0608) [inline code/game/objects/items/toys.dm]: This doesn't seem to do anything

**Comment by Jordie0608** [comment]
@Bawhoppen Doesn't matter who coded it first, niknak is porting it.

**Comment by Niknakflak** [comment]
The original thing defines the icon and cooldown

**Comment by Aranclanos** [comment]
werent they references to eachother because nt was like 10 people playing on the second server with their circlejerk

**Comment by Niknakflak** [comment]
I didnt make the original sprites aran.  Stop asking questions
>_>
<_<

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1212,4 +1212,207 @@
...
+	var/cooldown = 0
+	var/toysay = "What the fuck did you do?"
+
+/obj/item/toy/figure/New()
+    desc = "A \"Space Life\" brand [src]."
+
+/obj/item/toy/figure/attack_self(mob/user as mob)
+	if(cooldown <= world.time)
+		cooldown = world.time + 50
+		user << "<span class='notice'>The [src] says \"[toysay]\"</span>"
```
Comment (duncathan) [inline code/game/objects/items/toys.dm]: Should probably be `\The [src]`

**Comment by Dorsisdwarf** [comment]
The cargo tech doll is a reference but I dont remember who

**Comment by Niknakflak** [comment]
That is the most generic sprite there is.  Anyone could have short brown hair, are you kidding me?  Should I change the hair blue to appease you great ones

**Comment by Dorsisdwarf** [comment]
I meant the QM, not the ctech, whoops.

There was some shitty QM with a meme name who would deny all requests every round and had that black mohawk

**Comment by MrSnapwalk** [comment]
I can tell you with absolute certainty (because I made the sprites) that all cosmetic features were applied completely randomly. No references.

**Comment by Niknakflak** [comment]
The virologist totally isn't magni we swear

**Pull request merged**